File: cs_t4_full/Clone_1160.cs
public static int IndexOf (byte [] haystack, byte [] needle) {
    if (needle.Length == 0) {
        return 0;
    }
    int [] charTable = MakeCharTable (needle);
    int [] offsetTable = MakeOffsetTable (needle);
    for (int i = needle.Length - 1; i < haystack.Length;) {
        int j;
        for (j = needle.Length - 1; needle [j] == haystack [i]; -- i, -- j) {
            if (j == 0) {
                return i;
            }
        }
        i += Math.Max (offsetTable [needle.Length - 1 - j], charTable [haystack [i]]);
    }
    return - 1;
}


 public static int IndexOf (byte [] haystack, byte [] needle) {
    // handle null cases
    if (needle.Length == 0) {
        return 0;
    }

    // pre-process the needle
    int [] charTable = MakeCharTable (needle);
    int [] offsetTable = MakeOffsetTable (needle);

    // perform the search
    int haystackLength = haystack.Length;
    int needleLength = needle.Length;
    int i = needleLength - 1;
    while (i < haystackLength) {
        int j = needleLength - 1;
        while (needle[j] == haystack[i]) {
            if (j == 0) {
                return i;
            }
            i--;
            j--;
        }
        i += Math.Max(offsetTable[needleLength - 1 - j], 
            charTable[haystack[i]]);
    }
    return -1;
}

File: java_t3_full/Clone_1160.java
static int search (int [] A, int K) {
    int l = 0;
    int u = A.length - 1;
    int m;
    while (l <= u) {
        m = (u - l) / 2 + l;
        if (A [m] < K) l = m + 1;
        else if (A [m] == K) return m;
        else u = m - 1;
    }
    return - 1;
}


 static int search(int[] list, int K){
    int min = 0;
    int max = list.length - 1;
    int mid;
    while(min <= max){
        mid = (max - min) / 2 + min;
        if(list[mid] < K){
            min = mid + 1;
        }else if(list[mid] == K){
            return mid;
        }else{
            max = mid - 1;
        }
    }
    return -1;
}


